.github/workflows/on-trigger.yml

```diff
@@ -0,0 +1,147 @@
+# This workflow runs all workflows to build the dependencies required for the
```

Comment:
This workflow also highly duplicates the previous one and we should merge them in a single workflow

Reply:
FWIW I do agree with @bthomee - these two workflows have different responsibilities (if you look outside of build/test cycle) hence should be separate.